compose/utils.py

```diff
@@ -29,16 +28,16 @@ def parallel_execute(objects, obj_callable, msg_index, msg):
     q = Queue()
 
     def inner_execute_function(an_callable, parameter, msg_index):
+        error = None
         try:
             result = an_callable(parameter)
         except APIError as e:
-            errors[msg_index] = e.explanation
+            error = e.explanation
             result = "error"
-        except Exception as e:
-            errors[msg_index] = e
+        except Exception as error:
             result = 'unexpected_exception'
 
-        q.put((msg_index, result))
+        q.put((msg_index, result, error))
```

Comment:
Apparently this breaks in py3 with 

```
UnboundLocalError: local variable 'error' referenced before assignment
```

If I change the second except to `except Exception as e: error = e` it succeeds.

I guess there is something special about using a variable in the exception handler